hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/FrameRunner.java

```diff
@@ -167,6 +169,8 @@ private long effectiveGasUsed(final long gasLimit, @NonNull final MessageFrame f
     private void propagateHaltException(MessageFrame frame, ExceptionalHaltReason haltReason) {
         if (haltReason.equals(INSUFFICIENT_CHILD_RECORDS)) {
             setPropagatedCallFailure(frame, RESULT_CANNOT_BE_EXTERNALIZED);
+        } else if (haltReason.equals(INVALID_FEE_SUBMITTED)) {
```

Comment:
Question - is the propagation of the `INVALID_FEE_SUBMITTED` necessary for mono service fidelity?

Reply:
It is necessary for fixing the behaviour. This bug was found in mono but the fix will be only in modu codebase, from my understanding